### PR TITLE
pentagonal-pound: fix 404s on @user/team pages

### PR DESCRIPTION
**Pull request by taravancil**
## Links
* [pentagonal-pound.glitch.me](https://pentagonal-pound.glitch.me/)
* [Manuscript](https://glitch.fogbugz.com/f/cases/3328418/404-page-for-user-and-team-pages-is-broken)

This fixes broken 404 pages for @team/user pages.

Visit [pentagonal-pound.glitch.me/@404](https://pentagonal-pound.glitch.me/@404) to test

## GIF/Screenshots:

**Before**
image

**After**
image

### CODE REVIEW

File: src/presenters/pages/team-or-user.js [hunk 1]
```diff
@@ -65,7 +65,12 @@ const getTeam = async (api, name) => {
 
 const TeamPageLoader = ({ id, name, ...props }) => {
   const api = useAPI();
-  return <DataLoader get={() => getTeam(api, name)}>{(team) => (team ? <TeamPage team={team} {...props} /> : <NotFound name={name} />)}</DataLoader>;
```
Comment (taravancil) [inline src/presenters/pages/team-or-user.js]: This `<NotFound>` was never reached

File: src/presenters/pages/team-or-user.js [hunk 2]
```diff
@@ -75,7 +80,9 @@ TeamPageLoader.propTypes = {
 const UserPageLoader = ({ id, name, ...props }) => {
   const api = useAPI();
   return (
-    <DataLoader get={() => getUserById(api, id)}>{(user) => (user ? <UserPage user={user} {...props} /> : <NotFound name={name} />)}</DataLoader>
```
Comment (taravancil) [inline src/presenters/pages/team-or-user.js]: This `<NotFound>` was never reached

Comment (modernserf) [reply src/presenters/pages/team-or-user.js]: feel like we should also have 404s for https://pentagonal-pound.glitch.me/user/123332443234234

Comment (taravancil) [reply src/presenters/pages/team-or-user.js]: Updated

image

File: src/presenters/pages/team-or-user.js [hunk 3]
```diff
@@ -91,8 +94,8 @@ const TeamOrUserPageLoader = ({ name, ...props }) => {
         team ? (
           <TeamPage team={team} {...props} />
         ) : (
-          <DataLoader get={() => getUserByLogin(api, name)}>
-            {(user) => (user ? <UserPage user={user} {...props} /> : <NotFound name={name} />)}
+          <DataLoader get={() => getUserByLogin(api, name)} renderError={() => <NotFound name={'@' + name} />}>
```
Comment (taravancil) [inline src/presenters/pages/team-or-user.js]: Added the '@' to be consistent with [`routes.js`](https://github.com/FogCreek/Glitch-Community/blob/master/server/routes.js#L137)

File: src/presenters/pages/team-or-user.js [hunk 4]
```diff
@@ -91,8 +94,8 @@ const TeamOrUserPageLoader = ({ name, ...props }) => {
         team ? (
           <TeamPage team={team} {...props} />
         ) : (
-          <DataLoader get={() => getUserByLogin(api, name)}>
-            {(user) => (user ? <UserPage user={user} {...props} /> : <NotFound name={name} />)}
+          <DataLoader get={() => getUserByLogin(api, name)} renderError={() => <NotFound name={`@${name}`} />}>
```
Comment (taravancil) [inline src/presenters/pages/team-or-user.js]: Added the '@' to be consistent with [`routes.js`](https://github.com/FogCreek/Glitch-Community/blob/master/server/routes.js#L137)

**Pull request merged**